lib/puppet/provider/service/smf.rb

```diff
@@ -49,6 +49,52 @@ def self.instances
    end
   end
 
+  # Returns the service's FMRI. We fail if multiple FMRIs correspond to
+  # @resource[:name].
+  #
+  # If the service does not exist or we fail to get any FMRIs from svcs,
+  # this method will raise a Puppet::ExecutionFailure
+  def service_fmri
+    return @fmri if @fmri
+
+    # `svcs -l` is better to use because we can detect service instances
+    # that have not yet been activated or enabled (i.e. it lets us detect
+    # services that svcadm has not yet touched). `svcs -H -o fmri` is a bit
+    # more limited.
+    lines = svcs("-l", @resource[:name]).chomp.lines.to_a
+    lines.select! { |line| line =~ /^fmri/ }
+    fmris = lines.map! { |line| line.split(' ')[-1].chomp }
+    unless fmris.length == 1
+      raise Puppet::Error, _("Failed to get the FMRI of the %{service} service: The pattern '%{service}' matches multiple FMRIs! These are the FMRIs it matches: %{all_fmris}") % { service: @resource[:name], all_fmris: fmris.join(', ') }
+    end
+
+    @fmri = fmris.first
+  end
+
+  # Returns true if the provider supports incomplete services.
+  def supports_incomplete_services?
+    Puppet::Util::Package.versioncmp(Facter.value(:operatingsystemrelease), '11.1') >= 0
```

Comment:
11.1 was chosen since that is the lowest Solaris 11 VM we have. I don't know if 11.0 supports incomplete services. Note that 11.1 is enough to fix the customer's issue, which was caught on Sol 11.3.

Reply:
From a review of https://www.oracle.com/technetwork/articles/servers-storage-admin/intro-smf-basics-s11-1729181.html, it appears that 11.0 does support incomplete services...

Reply:
Makes perfect sense to me. Not confirmed is not confirmed.